pulp_docker/app/registry.py

```diff
@@ -131,45 +131,26 @@ def _base_paths(path):
         distribution = self._match_distribution(path)
         repository_version = distribution.get_repository_version()
         accepted_media_types = await Registry.get_accepted_media_types(request)
-        if MEDIA_TYPE.MANIFEST_LIST in accepted_media_types:
-            try:
-                tag = ManifestListTag.objects.get(
-                    pk__in=repository_version.content,
-                    name=tag_name
-                )
-            # If there is no manifest list tag, try again with manifest tag.
-            except ObjectDoesNotExist:
-                pass
-            else:
-                response_headers = {'Content-Type': MEDIA_TYPE.MANIFEST_LIST}
-                return await Registry.dispatch_tag(tag, response_headers)
-
-        if MEDIA_TYPE.MANIFEST_V2 in accepted_media_types:
-            try:
-                tag = ManifestTag.objects.get(
-                    pk__in=repository_version.content,
-                    name=tag_name,
-                    manifest__schema_version=2
-                )
-            except ObjectDoesNotExist:
-                pass
-            else:
-                response_headers = {'Content-Type': MEDIA_TYPE.MANIFEST_V2}
-                return await Registry.dispatch_tag(tag, response_headers)
+        tag = ManifestTag.objects.get(
+            pk__in=repository_version.content,
+            name=tag_name,
+        )
+        if tag.tagged_manifest.media_type in accepted_media_types:
+            response_headers = {'Content-Type': tag.tagged_manifest.media_type}
```

Comment:
i suggest to add a check here: if it is a schema1 then return in `content-type` a signed_v1 header. I am not sure exactly what are the implications but to my knowledge client checks if it is a signed manifest then it has some algorithm on how to find/validate that signature with the manifest.

Reply:
@asmacdo change of plans https://docs.docker.com/registry/spec/manifest-v2-2/#backward-compatibility Second last paragraph.

if available media_type is among accepted types and it is not equal to schema1 then return as is.
elif available media type is schem1a  then return signed_v1
else raise pathnotresolved and the warning log